### DISCUSSION TITLE
TypeError: 'NoneType' object is not callable

### DISCUSSION THREAD

**Discussion by Elektuur** [MrYsLab/telemetrix#80] [Q&A] [answered]
When I run the program below:

`from telemetrix  import telemetrix

board = telemetrix.Telemetrix()
rot_motor = board.set_pin_mode_stepper(interface=1, pin1=11, pin2=12)

def running_callback(data):
        global exit_flag
        exit_flag += 1

def dummy_callback(data):
        pass

exit_flag = 0
board.stepper_move_to(rot_motor, 10)
board.stepper_run(rot_motor, running_callback)
board.stepper_is_running(rot_motor, dummy_callback)
while exit_flag == 0:
        pass`

I get the following error:

`(.venv) jvl@raspberrypi:~/Diameter $ /home/jvl/Diameter/.venv/bin/python /home/jvl/Diameter/test.py
Telemetrix:  Version 1.39

Copyright (c) 2021 Alan Yorinks All Rights Reserved.

Opening all potential serial ports...
        /dev/ttyUSB0

Waiting 4 seconds(arduino_wait) for Arduino devices to reset...
Valid Arduino ID Found.
Arduino compatible device found and connected to /dev/ttyUSB0
Waiting for Arduino to reset
Reset Complete

Retrieving Telemetrix4Arduino firmware ID...
Telemetrix4Arduino firmware version: 5.3.3
Calback:  <function dummy_callback at 0x7ffee6a3f600>
Exception in thread Thread-2 (_reporter):
Traceback (most recent call last):
  File "/usr/lib/python3.11/threading.py", line 1038, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.11/threading.py", line 975, in run
    self._target(*self._args, **self._kwargs)
  File "/home/jvl/Diameter/.venv/lib/python3.11/site-packages/telemetrix/telemetrix.py", line 2519, in _reporter
    dispatch_entry(response_data)
  File "/home/jvl/Diameter/.venv/lib/python3.11/site-packages/telemetrix/telemetrix.py", line 2453, in _stepper_is_running_report
    cb(cb_list)
TypeError: 'NoneType' object is not callable`

I am using Visual Studio Code Version: 1.88.1 on a Raspberry Pi 5 with Bookworm OS 64 bit. With the Python extension v2024.4.1. The Pi is connected with USB to an Arduino Uno kloon with Telemetrix4Arduino firmware

What causes this error and how can I fix it?

**Comment by MrYsLab** [comment, marked as answer]
Thanks for reporting this. I corrected a bug in Telemetrix4Arduino. Version 5.4.0 should be available in a day or two from the Arduino library manager. 

If you do not wish to wait for Arduino to publish the update, you may also download the[ corrected file here](https://github.com/MrYsLab/Telemetrix4Arduino/blob/master/examples/Telemetrix4Arduino/Telemetrix4Arduino.ino).
